### PR TITLE
Fix futunn parser issue

### PR DESCRIPTION
**Pull request by Wsine**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/futunn/highlights
```

## 新RSS检查列表 / New RSS Script Checklist
  
- [ ] New Route
- [ ] Documentation
  - [x] CN
  - [ ] EN
- [x] 全文获取 fulltext
  - [x] Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] 日期和时间 date and time
  - [x] 可以解析 Parsed
  - [x] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added 
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/routes/futunn/highlights.js [hunk 1]
```diff
@@ -1,56 +1,69 @@
...
+                return true;
+            })
+            .map(async (info) => {
+                const title = info.title;
+                const itemUrl = info.url;
+
+                const cache = await ctx.cache.get(itemUrl);
+                if (cache) {
+                    return Promise.resolve(JSON.parse(cache));
+                }
```
Comment (TonyRL) [inline lib/routes/futunn/highlights.js:37]: Please use `ctx.cache.tryGet()` for caching.
Docs: https://docs.rsshub.app/joinus/quick-start.html#ti-jiao-xin-de-rsshub-gui-ze-bian-xie-jiao-ben-shi-yong-huan-cun

**Review by TonyRL — CHANGES_REQUESTED** [review]

File: lib/routes/futunn/highlights.js (hunk 1 above)
Comment (Wsine) [reply lib/routes/futunn/highlights.js]: Thanks. Revised to use this caching interface.

File: lib/routes/futunn/highlights.js [hunk 2]
```diff
@@ -1,56 +1,66 @@
...
+                        link: itemUrl,
+                        description,
+                        pubDate: timezone(parseDate(date, 'YYYY/MM/DD HH:mm'), +8),
+                    };
+
+                    return sg;
+                });
+
+                return Promise.resolve(single);
+            })
```
Comment (TonyRL) [inline lib/routes/futunn/highlights.js:58]:
```suggestion
            .map((info) =>
                ctx.cache.tryGet(info.url, async () => {
                    const response = await got({
                        url: info.url,
                        method: 'get',
                        headers: {
                            Referer: link,
                        },
                    });

                    const $ = cheerio.load(response.data);
                    const description = $('div.newsDetailBox div.main div.inner').html();
                    const date = $('div.timeBar').children().remove().end().text().trim();

                    info.link = info.url;
                    info.description = description;
                    info.pubDate = timezone(parseDate(date, 'YYYY/MM/DD HH:mm'), +8);

                    return info;
                })
            )
```

Comment (TonyRL) [reply lib/routes/futunn/highlights.js:58]: You can remove extra async, await and Promise.resolve().

**Review by TonyRL — CHANGES_REQUESTED** [review]

**Comment by Wsine** [comment]
I committed the suggested changes. Please check. thanks.

**Comment by Wsine** [comment]
@TonyRL any problem with current PR?

**Review by TonyRL — APPROVED** [review]

**Pull request merged**